core/blockstm/dag.go

```diff
@@ -74,13 +74,19 @@ func BuildDAG(deps TxnInputOutput) (d DAG) {
 func (d DAG) Report(out func(string)) {
 	roots := make([]int, 0)
 	rootIds := make([]string, 0)
+	rootIdMap := make(map[int]string, 0)
```

Comment:
Maybe consider preallocating the map with the size of `len(d.GetRoots())`?

Reply:
Done